src/k8s/cmd/k8s/k8s_set_test.go

```diff
@@ -79,6 +79,47 @@ func generateMapstructureTestCasesStringSlice(keyName string, fieldName string)
 	}
 }
 
+func generateMapstructureTestCasesMap(keyName string, fieldName string) []mapstructureTestCase {
+	return []mapstructureTestCase{
+		{
+			val:        fmt.Sprintf("%s=", keyName),
+			assertions: []types.GomegaMatcher{HaveField(fieldName, map[string]string{})},
+		},
+		{
+			val:        fmt.Sprintf("%s={}", keyName),
+			assertions: []types.GomegaMatcher{HaveField(fieldName, map[string]string{})},
+		},
+		{
+			val:        fmt.Sprintf("%s=k1=", keyName),
+			assertions: []types.GomegaMatcher{HaveField(fieldName, map[string]string{"k1": ""})},
+		},
+		{
+			val:        fmt.Sprintf("%s=k1=,k2=test", keyName),
+			assertions: []types.GomegaMatcher{HaveField(fieldName, map[string]string{"k1": "", "k2": "test"})},
+		},
+		{
+			val:        fmt.Sprintf("%s=k1=v1", keyName),
```

Comment:
interesting that this format is valid

Reply:
this is what we enable parsing by specifying the `utils.StringToStringMapHookFunc()` hook.